src/foam/u2/crunch/CapabilityFeatureView.js

```diff
@@ -57,5 +106,31 @@ foam.CLASS({
           .add(self.data.description || 'no description')
         .end();
     }
+  ],
+
+  listeners: [
+    {
+      name: 'daoUpdate',
+      code: function() {
+        this.userCapabilityJunctionDAO.find(
+          this.AND(
+            this.OR(
+              this.AND(
+                this.NOT(this.INSTANCE_OF(this.AgentCapabilityJunction)),
+                this.EQ(this.UserCapabilityJunction.SOURCE_ID, this.associatedEntity.id)
```

Comment:
Couldn't this clause be moved to the top-level AND?

Reply:
ya we can clean this a litte. Suggest:
```
this.userCapabilityJunctionDAO.find(
          this.AND(
	    this.EQ(this.UserCapabilityJunction.TARGET_ID, this.data.id),
            this.EQ(this.UserCapabilityJunction.SOURCE_ID, this.associatedEntity.id),
            this.OR(
              this.NOT(this.INSTANCE_OF(this.AgentCapabilityJunction)),
              this.AND(
                this.INSTANCE_OF(this.AgentCapabilityJunction),
                this.EQ(this.AgentCapabilityJunction.EFFECTIVE_USER, this.subject.user.id)
              )
            )
          )
```

Reply:
Using @Nauna predicate, the results are the same.